src/rust/src/decoder/mod.rs

```diff
@@ -86,10 +88,161 @@ impl dtvcc_ctx {
             ),
         }
     }
-    pub fn process_current_packet(&mut self, len: i32) {
+    /// Process current packet into service blocks
+    pub fn process_current_packet(&mut self, len: u8) {
+        let seq = (self.current_packet[0] & 0xC0) >> 6;
+        debug!(
+            "dtvcc_process_current_packet: Sequence: {}, packet length: {}",
+            seq, len
+        );
+        if self.current_packet_length == 0 {
+            return;
+        }
+
+        if self.current_packet_length != len as i32 {
+            // Is this possible?
+            self.reset_decoders();
```

Comment:
We should answer that question :-) I'd suppose that illegal, but I don't know. And if yes - is resetting the decoder the right way to go? 
Do we write something on the logs or just silently try to recover?

Reply:
I think I'll just remove this check. It's not needed. We will never reach this case due to this check, https://github.com/CCExtractor/ccextractor/blob/master/src/rust/src/decoder/mod.rs#L63